pages/api/interest-zones/[id]/[catID].ts

```diff
@@ -0,0 +1,43 @@
+import { zoneService, ZoneValidationError } from '../../../../services/zone-service';
+
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { withApiAuthRequired } from '@auth0/nextjs-auth0';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse<any>) {
+    switch (req.method) {
+        case 'PUT':
+            try{
+                const newCat = await zoneService.editCat(req.query.id, req.query.catID, req.body);
+                res.json(newCat);
+            } catch (error: any) {
+                if (error instanceof ZoneValidationError) {
+                    res.status(error.getStatus()).json({message: error.getErrorCode()});
+                }
+                else {
+                    res.status(404).json({message: error.message});
+                }
+            }
+            break;
+        case 'PATCH':
+            try {
+                const newSterilizedCat = await zoneService.changeToSterilized(req.query.id, req.query.catID, req.body);
```

Comment:
Renaming the above method `changeToSterilized` to `sterilizeCat` would be more in line with the other related methods.